modules/generate-reference-doc/src/main/scala/scala/cli/doc/GenerateReferenceDoc.scala

```diff
@@ -230,16 +340,40 @@ object GenerateReferenceDoc extends CaseApp[InternalDocOptions] {
       }
     }
 
-    for (c <- mainCommands.iterator)
-      addCommand(c)
-    b.append(
-      """## Hidden commands
-        |
-        |""".stripMargin
-    )
-    for (c <- hiddenCommands.iterator)
-      addCommand(c, additionalIndentation = 1)
+    if (onlyRestricted) {
+      val scalaCommands = commands.collect { case s: ScalaCommand[_] => s }
+      b.section("# `scala` commands")
+      // TODO add links to RFC
+      b.section(
+        "This document is a specification of `scala` runner.",
+        "For now it uses documentation specific to Scala CLI but at some point it may be refactor to provide more abstract documentation.",
```

Comment:
```suggestion
        "For now it uses documentation specific to Scala CLI but at some point it may be refactored to provide more abstract documentation.",
```